crypto/fipsmodule/service_indicator/service_indicator_test.cc

```diff
@@ -897,11 +944,29 @@ TEST_P(ECDSA_ServiceIndicatorTest, ECDSAKeyCheck) {
 
   int approved = AWSLC_NOT_APPROVED;
 
+  // Test service indicator approval for |EC_KEY_generate_key_fips| and
+  // |EC_KEY_check_fips|.
   bssl::UniquePtr<EC_KEY> key(EC_KEY_new_by_curve_name(ecdsaTestVector.nid));
   CALL_SERVICE_AND_CHECK_APPROVED(approved, ASSERT_TRUE(EC_KEY_generate_key_fips(key.get())));
   ASSERT_EQ(approved, ecdsaTestVector.key_check_expect_approved);
   CALL_SERVICE_AND_CHECK_APPROVED(approved, ASSERT_TRUE(EC_KEY_check_fips(key.get())));
   ASSERT_EQ(approved, ecdsaTestVector.key_check_expect_approved);
+
+
+  // Test running the EVP_PKEY_derive interfaces one by one directly, and check
+  // |EVP_PKEY_keygen| for approval at the end. |EVP_PKEY_keygen_init| should
+  // not be approved because they do not indicate an entire service has been
+  // done.
+  bssl::UniquePtr<EVP_PKEY_CTX> ctx(EVP_PKEY_CTX_new_id(EVP_PKEY_EC, nullptr));
+  EVP_PKEY *raw = nullptr;
+  ASSERT_TRUE(ctx);
+  ASSERT_TRUE(EVP_PKEY_keygen_init(ctx.get()));
+  ASSERT_TRUE(EVP_PKEY_CTX_set_ec_paramgen_curve_nid(ctx.get(), ecdsaTestVector.nid));
+  CALL_SERVICE_AND_CHECK_APPROVED(approved, ASSERT_TRUE(EVP_PKEY_keygen(ctx.get(), &raw)));
+  // Prevent memory leakage.
+  bssl::UniquePtr<EVP_PKEY> pkey(raw);
```

Comment:
if you declare raw as UniquePtr before using it, does it help instead of doing that right before releasing it?